cypress/plugins/pluginUtils.js

```diff
@@ -115,4 +115,37 @@ function generateIndexFile(path, outputObj) {
   }
 }
 
-module.exports = { getAndDereferenceOpenRpc, generateIndexFile };
+/**
+ * @function preprocessDeviceData
+ * @description Reads the device data JSON file and adds it to the config object.
+ * @param {string} config - The config object.
+ * @example
+ * preprocessDeviceData(config);
+ */
+
+function preprocessDeviceData(config) {
+  const deviceMac = config.env.deviceMac;
+  try {
+    if (!deviceMac) {
+      throw new Error('Device MAC address is required.');
```

Comment:
Will this cause the run to fail? I don't think we should not run if the test if this happens. We can still run some tests without a mac. Aren't there valid use cases with a mock config where we don't need a mac?